src/Tasks/GenerateResource.cs

```diff
@@ -995,7 +995,8 @@ private bool TransformResourceFilesUsingResGen(List<ITaskItem> inputsToProcess,
 
             // We need to do a whole lot of work to make sure that we're not overrunning the command line ... UNLESS
             // we're running ResGen 4.0 or later, which supports response files. 
-            if (!_resgenPath.Equals(Path.GetDirectoryName(NativeMethodsShared.GetLongFilePath(ToolLocationHelper.GetPathToDotNetFrameworkSdkFile("resgen.exe", TargetDotNetFrameworkVersion.Version35))), StringComparison.OrdinalIgnoreCase))
+            if (NativeMethodsShared.IsWindows &&
```

Comment:
It might be easier to read/understand if:

```
bool usingResgenWithResponseFileSupport = NativeMethodsShared.IsWindows &&
                                   !_resgenPath.Equals(Path.GetDirectoryName(NativeMethodsShared.GetLongFilePath(ToolLocationHelper.GetPathToDotNetFrameworkSdkFile("resgen.exe", TargetDotNetFrameworkVersion.Version35))), StringComparison.OrdinalIgnoreCase);

if (usingResgenWithResponseFileSupport)
{
    ...
```

And update the comment to mention that mono's resgen supports response file only on windows.